### PR TITLE
support port change in FS http server

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/actions/fs/index.js [hunk 1]
```diff
@@ -536,7 +537,13 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
         // but the path has been removed since then.
         return
       }
-      throw err
+      // It's still possible we have a critical error, but if it's just the
+      // server port number that's changed, it's hard to detect. So just treat
+      // all other errors as this case. If this is actaully a critical error,
```
Comment (jzila) [inline shared/actions/fs/index.js]: actually

File: shared/actions/fs/index.js [hunk 2]
```diff
@@ -536,7 +537,13 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
...
+      // we end up doing this 3 times for nothing, which isn't as bad as trump
+      // elected as POTUS.
```
Comment (jzila) [inline shared/actions/fs/index.js]: Amusing, but we shouldn't have "political" speech in our OSS codebase.

**Comment by songgao** [comment]
ready again!

File: shared/actions/fs/index.js [hunk 3]
```diff
@@ -514,8 +515,8 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
   let localHTTPServerInfo: Types._LocalHTTPServer =
     state.fs.localHTTPServerInfo || Constants.makeLocalHTTPServer()
   // This should finish within 2 iterations most. But just in case we bound it
```
Comment (jzila) [inline shared/actions/fs/index.js]: 2 iterations *at* most.

File: shared/actions/fs/index.js [hunk 4]
```diff
@@ -536,7 +537,13 @@ function* _loadMimeType(path: Types.Path, refreshTag?: Types.RefreshTag) {
...
         return
       }
-      throw err
+      // It's still possible we have a critical error, but if it's just the
+      // server port number that's changed, it's hard to detect. So just treat
+      // all other errors as this case. If this is actually a critical error,
+      // we end up doing this 3 times for nothing, which isn't the end of the
+      // world.
+      logger.info(`_loadMimeType i=${i} error:`, err)
+      localHTTPServerInfo.address = ''
```
Comment (jzila) [inline shared/actions/fs/index.js]: So this will trigger a re-establishment of the server?

Comment (songgao) [reply shared/actions/fs/index.js]: Hmm not sure what you mean by "re-establishment of the server", but on next iteration this causes a call to `SimpleFSSimpleFSGetHTTPAddressAndTokenRpcPromise`, which refreshes address (with port) and token.

**Pull request merged**